include/onnxruntime/core/optimizer/graph_transformer.h

```diff
@@ -9,15 +9,33 @@
 
 namespace onnxruntime {
 
+/**
+* Default_Global refers to all the transformations which are required for 
+  corrcetness and are execution provider independent
+* Default_ProvderSpecific refers to transformations which are also required 
+  for correctness but are execution provider specific
+* Optional_L1 refers to all basic optmizations. These are run both before and 
+  after graph partitioning
+* Optional_L2 refers to all fusion optmizations. These are run after graph 
+  partitioning
+*/
+enum TransformerLevel {
+  Default_Global = 0,
+  Default_ProviderSpecific = 1,
+  Optional_L1 = 2,
+  Optional_L2 = 4,  
+  MaxTransformerLevel = 8
+};
+
 /**
 @class GraphTransformer
 
 The interface for in-place transformation of a Graph.
 */
 class GraphTransformer {
  public:
-  GraphTransformer(const std::string& name, const std::string& desc)
-      : name_(name), desc_(desc) {
+  GraphTransformer(const std::string& name, const std::string& desc, const TransformerLevel level, std::vector<std::string>&& providers)
```

Comment:
I am not sure it is a good idea to hardcode optimization level within the GraphTransformer itself.
What if we later decide to change the level of a graph transformation or add a new (third and so on) level? Then we would have to go to each graph transformer class and change the hardcoded level.
Why not have a central place where we simply register the level of a graph transformer? Then we would just have to change it in that single file.